panic: runtime error: index out of range

Please answer the following before submitting your issue:

Note: Please include any substantial examples (debug session output,
stacktraces, etc) as linked gists.

1. What version of Delve are you using (`dlv version`)?
Delve Debugger
Version: 1.2.0
Build: $Id: 068e2451004e95d0b042e5257e34f0f08ce01466 $

2. What version of Go are you using? (`go version`)?
go version go1.10.4 linux/amd64

3. What operating system and processor architecture are you using?
```bash
# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.2 LTS
Release:        18.04
Codename:       bionic

# uname -a
Linux dave-desktop 4.15.0-46-generic #49-Ubuntu SMP Wed Feb 6 09:33:07 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```
4. What did you do?
I am trying to debug a static bianry, the debug used to be work, but doesn't work now, what I did is upgrade go to latest version, and I reverted it back after I saw the error. I can debug successfully with a simple go binary instead.


 ```# dlv exec ...```

5. What did you expect to see?
6. What did you see instead?
```
...
panic: runtime error: index out of range

goroutine 35 [running]:
github.com/go-delve/delve/pkg/proc.(*BinaryInfo).loadDebugInfoMapsCompileUnit(0xc420001380, 0xc42000c520, 0xc42022c000, 0xc4200663c0, 0xc42051c090, 0xc420520200)
        /go/src/github.com/go-delve/delve/pkg/proc/types.go:426 +0x173f
github.com/go-delve/delve/pkg/proc.(*BinaryInfo).loadDebugInfoMaps(0xc420001380, 0xc42022c000, 0xc4203b6000, 0x86bc, 0x86bc, 0xc4201c41b0, 0x0)
        /go/src/github.com/go-delve/delve/pkg/proc/types.go:266 +0x59e
created by github.com/go-delve/delve/pkg/proc.loadBinaryInfoElf
        /go/src/github.com/go-delve/delve/pkg/proc/bininfo.go:848 +0x39a
```

--- a/pkg/proc/types.go
+++ b/pkg/proc/types.go
@@ -263,7 +263,9 @@ func (bi *BinaryInfo) loadDebugInfoMaps(image *Image, debugLineBytes []byte, wg 
 				}
 			}
 			bi.compileUnits = append(bi.compileUnits, cu)
-			cu.endOffset = bi.loadDebugInfoMapsCompileUnit(ctxt, image, reader, cu)
+			if entry.Children {
+				cu.endOffset = bi.loadDebugInfoMapsCompileUnit(ctxt, image, reader, cu)
+			}
 
 		case dwarf.TagPartialUnit:
 			reader.SkipChildren()

